scripts/scr_poststage.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+#
+# SCR allows you to spawn off checkpoint transfers "in the background"
+# that will finish some time after a job completes.  This saves you from
+# using your compute time to transfer checkpoints.  You can do this by
+# specifying XFER=BBAPI_POSTSTAGE for a storage descriptor in scr.conf.
+# Currently, this is only supported on on IBM burst buffer nodes.
+#
+# This script is to be run as a 2nd-half post-stage script on an IBM
+# system.  A 2nd-half post-stage script will run after all the job's burst
+# buffers transfers have finished (which could be hours later after the job
+# finishes).  This script will finalize any completed burst buffer checkpoint
+# transfers so that they're visible to SCR.
+#
+
+# Please fill in BINDIR AND PREFIX below:
+
+# The dir path containing the SCR binaries (scr_index and scr_flush_file)
+BINDIR=""
+# The dir path to your SCR PREFIX
+PREFIX=""
+
+# Path to where you want the scr_poststage.sh log
+LOGFILE=/tmp/scr_poststage.log
+
+if [ ! -d "$BINDIR" ] ; then
+	echo "Please specify BINDIR in scr_poststage.sh" >> $LOGFILE
+	exit
+fi
+
+if [ ! -d "$PREFIX" ] ; then
+	echo "Please specify PREFIX in scr_poststage.sh" >> $LOGFILE
+	exit
+fi
+
+export PATH="$PATH:$BINDIR"
+
+echo "$(date '+%m/%d/%y %H:%m:%S') Begin post_stage" >> $LOGFILE
+echo "Current index before finalizing:" >> $LOGFILE
+scr_index -l -p $PREFIX >> $LOGFILE
+
+# Finalize each dataset listed in `scr_index -l -p` that is NOT
+# flushed.
+#
+# $ scr_index -l -p $PREFIX
+# DSET VALID FLUSHED             CUR NAME
+#   2 NO                          ckpt.2
+#   1 NO                          ckpt.1
```

Comment:
Let's use the flush file to find the set of datasets to finish, rather than the index file.  The flush file should only contain info for datasets that have been flushed in the current job.  The index file will also contain info for lots of older datasets written from earlier jobs.

Reply:
There are a lot of similarities in what we're doing in poststage with what we do in our scavenge operation.  As future work, we'll need to integrate poststage capability with scavenge.  That may be a lot of work, so we'll save that poststage+scavenge step for a future PR.

In the meantime, the logic to identify the datasets to scavenged is here:
https://github.com/LLNL/scr/blob/develop/scripts/common/scr_postrun.in#L112-L249

There are two loops there.  The goal of that code is to scavenge every dataset that is marked as output and also the most recent checkpoint or at least the most recent checkpoint that comes before the first failed dataset.

We could use similar scr_flush_file commands to get the dataset ids and names to wait on in the poststage script.  We are likely missing the info about whether the dataset was flushed with BBAPI.